File: front/src/app/_models/userParams.ts
import { MyUser } from "./MyUser";

export class UserParams
{
    minAge = 18;
    maxAge = 70;
    bloodGroup: string;
    city: string;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lastActive';

    constructor(user: MyUser)
    {
        this.bloodGroup = user.bloodGroup
        this.city = user.city
    }
}

File: back/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace back.Helpers
{
    public class UserParams
    {
        private const int MaxPageSize = 50;
        public int PageNumber { get; set; } = 1;
        private int _pageSize = 10; 

        public int PageSize
        {
            get => _pageSize;
            set => _pageSize = (value > MaxPageSize) ? MaxPageSize : value;
        }

        public string Email { get; set; }
        public string BloodGroup { get; set; }
        public string OrderBy { get; set; } = "lastActive";
        public string City { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 70;
        
    }
}